3.js

```diff
@@ -0,0 +1,20 @@
+function accessWordsByIndex (string, index) {
+  const arrayOfWords = splitStringIntoArrayOfWords(arrayToString(removePuncuationFromArray(stringToArray(string))))
```

Comment:
This line is of significant enough complexity that it might be an additional function? 
```js
const findWords = string => 
  [ splitIntoWords, arrayToString, removePunctuation stringToArray ].reduce( (current, fn) => fn( current ), string )

const accessWordsByIndex = ( string, index ) =>
  findWords( string )[ index - 1 ] || ''
```
Just brainstorming, not sure if I'm in love with that approach, but it does help clean up the function composition a little...